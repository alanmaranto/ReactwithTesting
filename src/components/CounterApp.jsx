import React from "react";
import PropTypes from "prop-types";

const CounterApp = ({ value }) => {
  const onAdd = (e) => {
    console.log(e);
  };
  return (
    <div>
      <h1>CounterApp</h1>
      <h2>{value}</h2>

      <button onClick={onAdd}>+1</button>
    </div>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number,
};

export default CounterApp;
