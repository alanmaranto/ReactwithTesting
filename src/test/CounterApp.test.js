import React from "react";
import { shallow } from "enzyme";
import CounterApp from "../components/CounterApp";

describe("Tests in counter App", () => {
  test("should display App Correctly", () => {
    const wrapper = shallow(<CounterApp />);

    expect(wrapper).toMatchSnapshot();
  });
  test('should display counter with value of 100', () => {
    const counter = 100;
    const wrapper = shallow(<CounterApp value={counter} />);

    const counterValue = '100';
    const counterText = wrapper.find('h2').text()
    expect(counterText).toBe(counterValue);
  })
  
});
