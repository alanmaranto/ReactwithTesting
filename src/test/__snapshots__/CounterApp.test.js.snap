// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in counter App should display App Correctly 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
    10
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</div>
`;
